### PR TITLE
ruleUID should be part of 'ctx'

### PR DESCRIPTION
**Pull request by maggu2810**
Fixes: https://github.com/eclipse/smarthome/issues/3590

@smerschjohann Could you have a look at?

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java [hunk 1]
```diff
@@ -106,10 +107,13 @@ private void loadConfig() {
...
 
-        // make the whole context available as "ctx"
+        // Add the rule's UID to the context
+        final Map<String, Object> contextNew = new HashMap<>(context);
+        contextNew.put("ruleUID", this.ruleUID);
+
+        // make the whole (new) context available as "ctx"
         // Note: We don't use "context" here as it doesn't work on all JVM versions!
-        executionContext.setAttribute("ctx", context, ScriptContext.ENGINE_SCOPE);
-        executionContext.setAttribute("ruleUID", this.ruleUID, ScriptContext.ENGINE_SCOPE);
```
Comment (triller-telekom) [inline bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java]: is it correct that you remove the `ruleUID` from the global context again?

I understand the [comment](https://github.com/eclipse/smarthome/issues/3590#issuecomment-312091953) from @smerschjohann in a way that it should be available in **both** contexts, i.e. the local and the global one.

Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java]: @triller-telekom You are right, I moved ruleUID from global namespace to ctx (so removed it from the global namespace).
#3007 contains the discussion to add ruleUID to "ctx".
#1783 moved ruleUID from "ctx" to the global namespace without any discussion or agreement (IMHO).
I assume (reading your comment) you are using "ruleUID" from the global now as I use it from "ctx".

You are right, the comment of @smerschjohann indicates that it should be also in the global namespace.

So, what should we do?

Comment (sjka) [inline bundles/automation/org.eclipse.smarthome.automation.module.script/src/main/java/org/eclipse/smarthome/automation/module/script/internal/handler/AbstractScriptModuleHandler.java]: Looks like we are back at the same point where also  https://github.com/eclipse/smarthome/pull/3152#issuecomment-287660146 got stuck. If we ever want to have a chance to offer a versioned "API" to scripts, then the global namespace would be poisonous in general. But the ruleUID in particular is so elemental that we can safely accept the "risk" of having it globally as well, IMHO.

**Comment by maggu2810** [comment]
> I understand the [comment](https://github.com/eclipse/smarthome/issues/3590#issuecomment-312091953) from @smerschjohann in a way that it should be available in **both** contexts, i.e. the local and the global one.

Changed.

**Pull request merged**